collection-scripts/gather

```diff
@@ -4,48 +4,61 @@ unset KUBECONFIG
 for localns in $(/usr/bin/oc get migrationcontrollers.migration.openshift.io --all-namespaces --no-headers | awk '{print $1}'); do
   clusters=()
   clusters+=(host)
+  # get all migclusters
   for c in $(oc -n ${localns} get migclusters -o custom-columns=name:.metadata.name --no-headers); do
+    # only take non-host cluster
     if [ $(oc -n ${localns} get migcluster ${c} -o go-template='{{ .spec.isHostCluster }}') == "false" ]; then
       clusters+=(${c})
     fi
   done
+  echo "Will collect debug info from migclusters [${clusters[@]}]"
 
+  # Iterate over all connected non-host OpenShift clusters
   for cluster in ${clusters[@]}; do
     unset KUBECONFIG
     if [ ${cluster} != "host" ]; then
+      echo "[cluster=${cluster}] Performing token login"
       token=$(oc get -n $(oc get -n ${localns} migcluster ${cluster} -o go-template='{{ .spec.serviceAccountSecretRef.namespace }}') secret $(oc get -n ${localns} migcluster ${cluster} -o go-template='{{ .spec.serviceAccountSecretRef.name }}') -o go-template='{{ .data.saToken }}' | base64 -d)
       url=$(oc get -n ${localns} migcluster ${cluster} -o go-template='{{ .spec.url }}')
       export KUBECONFIG=/tmp/kubeconfig
       oc login --insecure-skip-tls-verify=true --token=${token} $url
       if [ "$?" != "0" ]; then
         object_collection_path="/must-gather/${cluster}/"
         mkdir -p ${object_collection_path}
-        echo "login to cluster ${cluster} unsuccessful."  > "${object_collection_path}/login_error.txt"
+        echo "Login to migcluster ${cluster} unsuccessful."  > "${object_collection_path}/login_error.txt"
         continue
       fi
     fi
 
-    # Migration namespace list
+    # Build list of namespaces where MTC is installed
     namespaces=()
-
     for ns in $(/usr/bin/oc get migrationcontrollers.migration.openshift.io --all-namespaces --no-headers | awk '{print $1}')
     do
-      namespaces+=(ns/$ns)
+      echo "[cluster=${cluster}][namespace=${ns}] Detected MTC installation"
+      namespaces+=(${ns})
     done
 
-    # Run the collection of resources using must-gather
+    # Collect resources in MTC namespaces with must-gather
     for ns in ${namespaces[@]}; do
-      /usr/bin/oc adm inspect --dest-dir must-gather/${cluster} --all-namespaces ${ns}
+      echo "[cluster=${cluster}][namespace=${ns}] Running oc adm inspect"
+      /usr/bin/oc adm inspect --dest-dir must-gather/${cluster} --all-namespaces ns/${ns} &
     done
 
-  done
-  # Collect the migration and velero CRs
-  /usr/bin/gather_crs
+    # Collect the migration and velero CRs
+    echo "[cluster=${cluster}] Gathering MTC and Velero CRs for namespaces [${namespaces[@]}]"
+    /usr/bin/gather_crs ${cluster} ${namespaces} &
```

Comment:
Anywhere I thought it was safe to launch parallel background tasks, then to wait on them to complete before proceeding, I did it.